transcom/mymove
# Int b 21507 pod poe too validation

## [B-21507](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1018854)

## Summary

We are adding a view for the Port of Embark and Port of Debark for the TOO and any other office user with the ability to view the MTO page.

They should be able to view the Port of Embark or Port of Debark based on whichever one is populated the other port should be blank. Here is an example:

![image](https://github.com/user-attachments/assets/ff735551-f3f8-4497-9d12-f7ab7a799586)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- 
- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

> **_NOTE:_**  We are following the same steps for 1-5 from [B-21505](https://github.com/transcom/mymove/pull/14459) to setup the location for the POE/POD.

1. Create an INTL UB shipment as a customer (CONUS -> AK or AK -> CONUS)
2. Manually create the mto_service_items table for the shipment: `
INSERT INTO mto_service_items (
					id,
                    mto_shipment_id,
                    move_id,
                    re_service_id,
                    status,
                    created_at,
                    updated_at,
                    approved_at
                )
                VALUES (
                uuid_generate_v4(),
                    '8eeba537-bb40-4f58-a534-74ea9ff64590',
                    '813b7a43-2e3f-42f3-8c4e-eca95390d5ca',
                    '388115e8-abe9-441d-96cf-a39f24baa0a3',
                    'APPROVED'::service_item_status,
                    NOW(),
                    NOW(),
                    NOW()
                );
`
3. In the prime v3 api, call getMoveTaskOrder using the move_id for the service item you created in step 2. You will need the etag for the newly created service item for the next step.
4. In the prime v1 api, call updateMTOServiceItem using a payload like this (adjust for you specific service item ID): `
{
  "id": "40b7e71a-6b03-4413-8601-dea694253ee3",
  "modelType": "UpdateMTOServiceItemInternationalPortFSC",
  "portCode": "PDX",
  "reServiceCode": "PODFSC"
}
`
5. Verify that you receive a 200 response like this: `{
  "reServiceCode": "PODFSC",
  "eTag": "MjAyNC0xMi0wMlQxNjo0MjoxNi42NjUwNDla",
  "id": "40b7e71a-6b03-4413-8601-dea694253ee3",
  "modelType": "MTOServiceItemBasic",
  "moveTaskOrderID": "813b7a43-2e3f-42f3-8c4e-eca95390d5ca",
  "mtoShipmentID": "8eeba537-bb40-4f58-a534-74ea9ff64590",
  "reServiceName": "International POD Fuel Surcharge",
  "status": "APPROVED"
}`

6. Login as a TOO user or any office user that has permissions to access the MTO page. Search up the move id that we used to setup the POD/POE. When you travel to the MTO Page, you should now see the Port of Embark or Debark populated with the Port Code, Port Name, City, State, and Zip associated with it. 

### Frontend


### Backend


## Screenshots

![image](https://github.com/user-attachments/assets/d8ea9d15-025d-4877-a357-f1823b5e43d6)
